src/App.tsx

```diff
@@ -1,15 +1,47 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
+import debounce from 'lodash.debounce';
 import { peopleFromServer } from './data/people';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [chosenPerson, setChosenPerson] = useState('');
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, 300), []);
```

Comment:
You are using lodash's debounce function here, but it's not handling the case when a user enters only spaces. In such a case, the filter should not be called. You can add a check to ignore input that consists only of spaces.